File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        sub_paths = []
        split_paths = path.split('/')

        for split_p in split_paths:
            if len(split_p) == 0 or split_p == ".":
                continue
            elif split_p == "..":
                if len(sub_paths) > 0:
                    sub_paths.pop()
            else:
                sub_paths.append(split_p)
        
        ans = ""
        for sub_path in sub_paths:
            ans += "/" + sub_path
        if len(ans) == 0:
            return "/"
        return ans

File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> sub_paths;
        string sub_path = "";

        for (auto & i: path)
        {
            if (i=='/')
            {
                if (sub_path.length()!=0)
                {
                    if (sub_path.length()==1 && sub_path=="."){}
                    else if (sub_path.length()==2 && sub_path=="..")
                    {
                        if (sub_paths.size()>0)
                        {
                            sub_paths.pop_back();
                        }
                    }
                    else
                    {
                        sub_paths.push_back(sub_path);
                    }
                    sub_path = "";
                }
            }
            else
                sub_path += i;
        }

        if (sub_path.length()==1 && sub_path=="."){}
        else if (sub_path.length()==2 && sub_path=="..")
        {
            if (sub_paths.size()>0)
                sub_paths.pop_back();
        }
        else if (sub_path.length()>0)
            sub_paths.push_back(sub_path);

        string ans = "/";
        for (auto & s: sub_paths)
        {
            cout << s << endl;
            ans += s + "/";
        }

        if (ans.length()>1)
            return ans.substr(0, ans.size()-1);
        return ans;
    }
};